package-drone/src/main/java/de/dentrassi/pm/jenkins/DroneRecorder.java

```diff
@@ -485,19 +433,40 @@ public Void invoke ( final File basedir, final VirtualChannel channel ) throws I
                             filename = f;
                         }
                         uploader.upload ( file, filename );
-//                        TODO this.artifacts.put ( artifacId, f );
                     }
-
+                    uploadResult.addArtifacts ( uploader.complete () );
```

Comment:
> Complete finalized the transaction
It's difficult to think a transaction in a rest client and it could be treated.
What do you think about
* addArtifact collect the files to send
* upload or performUpload or run execute the real upload and throws exception
* getUploadedArtifacts
* close clone open connection
It's a little bit clear